src/overlays/diff_overlay.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+
+import {
+    DagreGraph,
+    JsonSDFG,
+    Point2D,
+    SDFVSettings,
+    SimpleRect,
+} from '../index';
+import {
+    GraphElementInfo,
+    SDFGElementGroup,
+    SDFGRenderer,
+} from '../renderer/renderer';
+import {
+    NestedSDFG,
+    SDFGNode,
+    SDFGElement,
+    SDFGElementType,
+    Edge,
+} from '../renderer/renderer_elements';
+import { DiffMap } from '../sdfg_diff_viewr';
+import { GenericSdfgOverlay, OverlayType } from './generic_sdfg_overlay';
+
+export class DiffOverlay extends GenericSdfgOverlay {
+
+    public static readonly type: OverlayType = OverlayType.BOTH;
+    public readonly olClass: typeof GenericSdfgOverlay = DiffOverlay;
+
+    private readonly CHANGED_COLOR = 'orange';
+    private readonly ADDED_COLOR = 'green';
+    private readonly REMOVED_COLOR = 'red';
+
+    public constructor(
+        renderer: SDFGRenderer,
+        private readonly diffMap?: DiffMap
+    ) {
+        super(renderer);
+
+        this.refresh();
+    }
+
+    public refresh(): void {
+        this.renderer.draw_async();
+    }
+
+    public shadeElem(
+        elem: Edge | SDFGNode, ctx: CanvasRenderingContext2D
+    ): void {
+        if (this.diffMap?.addedKeys.has(elem.attributes().guid))
```

Comment:
You should use the `guid()` method of SDFGElement here instead (and update it to prioritize the guid attribute if exists).

Reply:
Added